crates/core/src/operations/transaction/mod.rs

```diff
@@ -407,15 +427,20 @@ impl<'a> CommitBuilder {
         table_data: Option<&'a dyn TableReference>,
         log_store: LogStoreRef,
         operation: DeltaOperation,
-    ) -> Result<PreCommit<'a>, CommitBuilderError> {
-        let data = CommitData::new(self.actions, operation, self.app_metadata)?;
-        Ok(PreCommit {
+    ) -> PreCommit<'a> {
+        let data = CommitData::new(
+            self.actions,
+            operation,
+            self.app_metadata,
+            self.app_transaction,
+        );
+        PreCommit {
             log_store,
             table_data,
             max_retries: self.max_retries,
             data,
             post_commit_hook: self.post_commit_hook,
-        })
+        }
```

Comment:
this did not need to be fallible - a lot of the touched files just updates from this change.